File: Backend/JAVA/application/src/main/java/com/superhero/domain/model/Biography.java
package com.superhero.domain.model;

import java.io.Serializable;
import java.util.List;

import com.fasterxml.jackson.annotation.JsonProperty;

public final class Biography implements Serializable{

    @JsonProperty("full-name")
    public String fullName;

    @JsonProperty("alter-egos")
    public String alterEgos;

    @JsonProperty("aliases")
    public List<String> aliases;

    @JsonProperty("place-of-birth")
    public String placeOfBirth;

    @JsonProperty("first-appearance")
    public String firstAppearance;

    @JsonProperty("publisher")
    public String publisher;

    @JsonProperty("alignment")
    public String alignment;

    public Biography() {
    }

    public String getFullName() {
        return fullName;
    }

    public void setFullName(String fullName) {
        this.fullName = fullName;
    }

    public String getAlterEgos() {
        return alterEgos;
    }

    public void setAlterEgos(String alterEgos) {
        this.alterEgos = alterEgos;
    }

    public List<String> getAliases() {
        return aliases;
    }

    public void setAliases(List<String> aliases) {
        this.aliases = aliases;
    }

    public String getPlaceOfBirth() {
        return placeOfBirth;
    }

    public void setPlaceOfBirth(String placeOfBirth) {
        this.placeOfBirth = placeOfBirth;
    }

    public String getFirstAppearance() {
        return firstAppearance;
    }

    public void setFirstAppearance(String firstAppearance) {
        this.firstAppearance = firstAppearance;
    }

    public String getPublisher() {
        return publisher;
    }

    public void setPublisher(String publisher) {
        this.publisher = publisher;
    }

    public String getAlignment() {
        return alignment;
    }

    public void setAlignment(String alignment) {
        this.alignment = alignment;
    }
}

File: Backend/Typescript/src/domain/entities/Biography.ts
import { Expose } from "class-transformer";

export class Biography {
    @Expose({ name: "full-name" })
    public FullName: string | undefined;

    @Expose({ name: "alter-egos" })
    public AlterEgos: string | undefined;

    @Expose({ name: "aliases" })
    public Aliases: string[] | undefined;

    @Expose({ name: "place-of-birth" })
    public PlaceOfBirth: string | undefined;

    @Expose({ name: "first-appearance" })
    public FirstAppearance: string | undefined;

    @Expose({ name: "publisher" })
    public Publisher: string | undefined;

    @Expose({ name: "alignment" })
    public Alignment: string | undefined;

}